### PR TITLE
Smokeping improvement + script to generate config

### PR DESCRIPTION
**Pull request by laf**
The script to generate was kindly done by @Rosiak and included in this PR with permission.

Docs included on how to use + older smokeping setup.

### CODE REVIEW

File: doc/Extensions/Smokeping.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+```bash
+@include /opt/smokeping/etc/librenms.conf
+```
+
+Then you need to generate the config file (maybe even add a cron to schedule this in and reload smokeping). We've assumed a few locations for smokeping, the config file you want 
+to call it and where LibreNMS is:
+
+```bash
+cd /opt/librenms/scripts/
+php ./gen_smokeping.php > /opt/smokepgin/etc/librenms.conf
```
Comment (SaaldjorMike) [inline doc/Extensions/Smokeping.md]: Looks like a typo in the path to the destination.

File: doc/Extensions/Smokeping.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+/opt/smokeping/bin/smokeping --reload
+```
+
+Sample cron:
+
+```bash
+15   0    * * * root cd /opt/librenms/scripts && php ./gen_smokeping.php > /opt/smokepgin/etc/librenms.conf && /opt/smokeping/bin/smokeping --reload >> /dev/null 2>&1
```
Comment (SaaldjorMike) [inline doc/Extensions/Smokeping.md]: Also typo here.

**Comment by laf** [comment]
Thanks @SaaldjorMike - fixed.

**Comment by f0o** [comment]
trigger ci

**Comment by paulgear** [comment]
I'd like to see the 5-line block starting with "if ($config['smokeping']['integration'] === true) {" moved into a common function.

**Comment by laf** [comment]
Updated

**Comment by paulgear** [comment]
There's no need for generate_smokeping_dir() - it can be implemented by making $file on generate_smokeping_file() default to the empty string.

**Comment by laf** [comment]
Updated now to use just one function.

**Comment by paulgear** [comment]
Scrutinizer isn't happy with that.  I think you just want: 

function generate_smokeping_file($device, $file = '') {
    global $config;
    if ($config['smokeping']['integration'] === true) {
        return $config['smokeping']['dir'] .'/'. $device['type'] .'/'.$file;
    } else {
        return $config['smokeping']['dir'] . '/'.$file;
    }
}

**Comment by laf** [comment]
Updated :)

**Comment by laf** [comment]
Rebased.

**Comment by paulgear** [comment]
There's one thing Scrutinizer's unhappy with; I'll have a look at it if I get a chance today.

**Comment by laf** [comment]
Updated

**Comment by laf** [comment]
Bump

**Comment by laf** [comment]
Rebased

Bump

**Pull request merged**